lib/ar2dto/has_dto.rb

```diff
@@ -22,17 +22,52 @@ def has_dto
       end
 
       # @api public
-      def to_dto
-        all.map(&:to_dto)
+      def to_dto(options = {})
+        all.map { |record| record.to_dto(options) }
       end
     end
 
     # Wrap the following methods in a module so we can include them only in the
     # ActiveRecord models that declare `has_dto`.
     module InstanceMethods
       # @api public
-      def to_dto
-        "#{self.class.name}DTO".constantize.new(attributes)
+      def to_dto(options = {})
+        options = apply_global_configs(options)
+
+        hash = serializable_hash(options&.except(:include))
+        hash = add_associations(hash, options&.dig(:include))
+
+        "#{self.class.name}DTO".constantize.new(hash)
+      end
+
+      private
+
+      def apply_global_configs(options)
+        options[:except] = AR2DTO::Config.instance.except + Array(options[:except])
```

Comment:
Just to check, isn't there another option needed to be considered here, like for example the `only` option?

Reply:
yes, but it hasn't been implemented yet